tools/gui/tools_gui.py

```diff
@@ -0,0 +1,135 @@
+'''
+import Tkinter as tk
+import ttk, tkFileDialog
+#Importing the plotter backend. 
+import sys, os
+sys.path.append(os.path.join(os.path.dirname(__file__),"plotter"))
+import plotter.plotter_backend as plotter_tool
+
+class multiToolGUI:
+  '''
+  A class used to represent the GUI for the plotter (and future tools).
+
+  Methods
+  -------
+  select_file()
+    Gets the user chosen filename and filepath and creates widgets for the
+    user to choose their graph labels in a drop-down menu
+  submit_graph_options()
+    Retrieves user-selected graph labels and plots a graph
+  setIcon()
+    Sets the GUI favicon
+  '''
+
+  def __init__(self):
+    self.main_window = tk.Tk() #Creates the application window
+    self.main_window.title("MST Multirotor Multitool")
+    self.icon_name = 'ninja_icon.gif'
+    self.setIcon(self.icon_name)
+
+    self.notebook = ttk.Notebook(self.main_window) #Allows the creation of tabs
+    self.notebook.grid(columnspan=2, row=0)
+
+    #Positioning first tab widgets
+    self.plotter_frame = ttk.Frame(self.main_window)
+    self.plotter_label = ttk.Label(self.plotter_frame,
+                                   text="Welcome to the Multirotor Plotter!")
+    self.file_selector = ttk.Button(self.plotter_frame, text="Open .csv file",
+                                    command=self.select_file)
+
+    #Positioning contents of first tab
+    self.plotter_frame.grid(columnspan=2, row=0)
+    self.plotter_label.grid(columnspan=2, row=0)
+    self.file_selector.grid(columnspan=2, row=1)
+
+    #Creates the actual tab
+    self.notebook.add(self.plotter_frame, text="Plotter")
+
+    '''
+    Second tab contents for future tool.
+    #TODO Make the drone swarm controller simulator for the second tab
+    self.tab_2_frame = ttk.Frame(self.main_window)
+    self.tab_2_label = tk.Label(self.tab_2_frame,
+                                    text="WIP: Reserved for second tool")
+
+    #Positioning second tab widgets
+    self.tab_2_frame.grid(column=0, row=0)
+    self.tab_2_label.grid(column=0, row=0)
+
+    self.notebook.add(self.tab_2_frame, text="WIP Tool")
+    '''
+
+    self.main_window.mainloop() #Starts the GUI event loop
+
+  def select_file(self):
+    '''
+    Gets the filename and path of a user-selected file.
+    Then it creates two pull-down menus for the user to chose graph labels.
+    '''
+    self.csv_filepath = tkFileDialog.askopenfilename(filetypes=[
+                                                     ("CSV Files", "*.csv"),
+                                                     ("All files","*.*")])
+
+    if self.csv_filepath:
+      print(self.csv_filepath)
+      #Plotter backend retrieves list of possible graph labels
+      self.column_headers = plotter_tool.get_csv_headers(self.csv_filepath)
+      #Creates pull-down menus with graph label options and a submit button
+      self.column_options_1 = ttk.Combobox(self.plotter_frame,
```

Comment:
Does not follow DRY. Could easily write code to do this like so:

```Python
for column_index in range(NUM_COLUMNS):
    attr_name = 'column_options_{}'.format(column_index)
    combobox = ttk.Combobox(self.plotter_frame, values=self.column_headers)
    setattr(self, attr_name, combobox)
    getattr(self, attr_name).grid(column=column_index, row=2)
```

Reply:
So for now there will be at max 2 column headers (since the plotter only graphs one x-axis and one y-axis variable right now). I've followed your suggestion and have made two column_options with your code. I will have to get the value of these two headers and then pass them through a function, so can I reference these two variables directly?
`
plotter_tool.submit_chosen_columns(self.csv_filepath, self.column_options1, self.column_options2)
`
Or is there a better way of doing this because I am using these variables as if I explicitly named them?